src/main/java/org/folio/circulation/domain/ChangeDueDateRepository.java

```diff
@@ -0,0 +1,42 @@
+package org.folio.circulation.domain;
+
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.Result;
+import org.folio.circulation.support.ServerErrorFailure;
+
+import java.lang.invoke.MethodHandles;
+import java.util.concurrent.CompletableFuture;
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ChangeDueDateRepository {
+
+  private static final Logger log = LoggerFactory
+    .getLogger(MethodHandles.lookup().lookupClass());
+
+  private final LoanRepository loanRepository;
+
+  public ChangeDueDateRepository(Clients clients) {
+    loanRepository = new LoanRepository(clients);
+  }
+
+  public CompletableFuture<Result<Loan>> changeDueDate(String loanId,
+    DateTime dueDate) {
+
+    log.info("Changing due date for {} to {}", loanId, dueDate);
+    try {
+      return loanRepository.getById(loanId)
+        .thenApply(r -> r.map(loan -> loan.changeDueDate(dueDate)))
+        .thenApply(r -> r.map(loan -> {
+          loan.changeAction(LoanAction.DUE_DATE_CHANGE);
+          return loan;
+        }))
```

Comment:
Could these two operations be combined together? Why would it be useful to change the due date itself separately to recording the action?

Reply:
No because change due date is used in renewal, there is another action for that.

Reply:
Ok, that means there probably needs to be some internal changes to the loan class